polkadot/node/subsystem-types/src/messages.rs

```diff
@@ -207,16 +208,19 @@ pub enum CollatorProtocolMessage {
 	/// This should be sent before any `DistributeCollation` message.
 	CollateOn(ParaId),
 	/// Provide a collation to distribute to validators with an optional result sender.
-	/// The second argument is the parent head-data hash.
-	///
-	/// The result sender should be informed when at least one parachain validator seconded the
-	/// collation. It is also completely okay to just drop the sender.
-	DistributeCollation(
-		CandidateReceipt,
-		Hash,
-		PoV,
-		Option<oneshot::Sender<CollationSecondedSignal>>,
-	),
+	DistributeCollation {
+		/// The receipt of the candidate.
+		candidate_receipt: CandidateReceipt,
+		/// The hash of the parent head-data.
+		parent_head_data_hash: Hash,
```

Comment:
Do we still need this here if we always have the `parent_head_data` ?

Reply:
Only to avoid computing the hash twice. Can remove it if this is deemed to be a premature optimization.

Reply:
Fair, let's just put that in a comment for clarity.